File: problems/hogwarts_towers/submissions/accepted/hogwarts_towers.py
mountains = []
altitude = 9
result = ''
lost = False

mountains.extend(map(int, input().split(" ")))
# # game loop
for _ in range(8):
    # destroy the highest tower
    index_max = max(range(len(mountains)), key=mountains.__getitem__)
    # The index of the tower to fire on.
    result += str((index_max)+1) + " "
    # look ahead
    lookAhead = max(range(len(mountains)), key=mountains.__getitem__)
    if(mountains[lookAhead] > altitude):
        lost = True
        break
    # crash the tower
    mountains[index_max] = 0
    altitude -= 1
if lost:
    print("Hogwarts is down")
else:
    print(result)

File: problems/hogwarts_towers/submissions/accepted/hogwarts_towers.cpp
#include <iostream>
#include <algorithm>
#include <string>

using namespace std;

int main()
{
	int mountains[8];
	string result = "";
	bool lost = false;
	int altitude = 9;

	for(int i=0;i<8;i++){
		cin >> mountains[i];
	}
	for(int i=0;i<8;i++){
		int index_max =max_element(mountains,mountains+8)-mountains;

		result += to_string(index_max+1) +" ";

		if(mountains[index_max] > altitude){
			lost = true;
        	break;
		}

		mountains[index_max] = 0;
		altitude -= 1;

	}
	if(lost){
		cout << "Hogwarts is down" << endl;
	}else{
		cout << result << endl;
	}

	return 0;
}